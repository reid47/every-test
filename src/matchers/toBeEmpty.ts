export function toBeEmpty(received) {
  const pass = received.innerHTML === '';

  const message = pass
    ? 'Expected element not to be empty, but it was.'
    : `Expected element to be empty, but it has content: ${received.innerHTML}`;

  return { pass, message };
}
