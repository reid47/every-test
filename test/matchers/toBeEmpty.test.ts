import toBeEmpty from '../../src/matchers/toBeEmpty';

describe('toBeEmpty', () => {
  test('passing', () => {
    const el = document.createElement('div');
    el.innerHTML = '';

    const result = toBeEmpty(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('div');
    el.innerHTML = '<div></div>';

    const result = toBeEmpty(el);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
