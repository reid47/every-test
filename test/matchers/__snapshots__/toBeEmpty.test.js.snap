// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeEmpty failing 1`] = `"Expected element to be empty, but it has content: <div></div>"`;

exports[`toBeEmpty passing 1`] = `"Expected element not to be empty, but it was."`;
